File: Solutions/807. Max Increase to Keep City Skyline/807. Max Increase to Keep City Skyline.cpp
class Solution {
public:
    int maxIncreaseKeepingSkyline(vector<vector<int>>& grid) {
        int m=grid.size(),n=grid[0].size();
        vector<int> skh(m),skv(n);
        for(int i=0;i<m;++i)
            for(int j=0;j<n;++j){
                skh[i]=max(skh[i],grid[i][j]);
                skv[j]=max(skv[j],grid[i][j]);
            }
        int res=0;
        for(int i=0;i<m;++i)
            for(int j=0;j<n;++j)
                res+=min(skh[i],skv[j])-grid[i][j];
        return res;
    }
};

File: Solutions/807. Max Increase to Keep City Skyline/807. Max Increase to Keep City Skyline.py
class Solution:
    def maxIncreaseKeepingSkyline(self, grid: List[List[int]]) -> int:
        m,n,res=len(grid),len(grid[0]),0
        skh,skv=[0]*m,[0]*n
        for i in range(m):
            for j in range(n):
                skh[i]=max(skh[i],grid[i][j])
                skv[j]=max(skv[j],grid[i][j])
        for i in range(m):
            for j in range(n):
                res+=min(skh[i],skv[j])-grid[i][j]
        return res